react-form-controls/src/index.jsx

```diff
@@ -0,0 +1,38 @@
+/* eslint-disable no-console */
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+class NewsletterForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      value: ''
+    };
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleChange() {
+    this.setState({ value: event.target.value });
+  }
+
+  handleSubmit() {
+    event.preventDefault();
+    console.log(this.state);
+  }
+
+  render() {
+    return (
+      <form onSubmit={this.handleSubmit}>
+        Email
+        <input type="email" onChange={this.handleChange} />
```

Comment:
The input should also be passed a `value` prop so that the input's value is dictated by the state of the component.